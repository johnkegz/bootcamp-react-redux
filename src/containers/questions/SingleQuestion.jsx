import React from "react";
import { SingleQuestionAction } from "../../actions/SingleQuestionAction";
import { connect } from "react-redux";
export class SingleQuestion extends React.Component {
  componentDidMount() {
    this.props.SingleQuestionAction(this.props.match.params.id);
  }
  showEditData() {
    if (this.props.match.params.id) {
      return Array.from(this.props.ViewSingleQuestion.item).map((data) => {
        const question = data.questions;
        const ans = data.answers;
        const answerElements = ans.map((answer, i) => {
          return <p key={i}>{answer.answer}</p>;
        });
        return (
          <div key={data.question_id} className="txt-s">
            {question}
            {answerElements}
          </div>
        );
      });
    }
  }
  render() {
   
    return (
      <div className="body-container">
        {this.showEditData()}
      </div>
    );
  }
}
export const mapStateToProps = state => {
  return {
    ViewSingleQuestion: state.singleQuestion
  };
};

export default connect(
  mapStateToProps,
  { SingleQuestionAction }
)(SingleQuestion);
