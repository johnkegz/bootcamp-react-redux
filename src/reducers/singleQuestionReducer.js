import { SINGLE_QUESTION } from "../actions/types";
  const initialState = {
    item: {},
    
  };
  
  export default function(state = initialState, action) {
    switch (action.type) {
      case SINGLE_QUESTION:
        return {
          ...state,
          item: action.payload
        };
      
      default:
        return state;
    }
  }
  