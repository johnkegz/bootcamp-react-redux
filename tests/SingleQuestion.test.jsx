import React from "react";
import { shallow } from "enzyme";
import SingleQuestion from "../src/containers/questions/SingleQuestion";
import singleQuestionReducer from "../src/reducers/singleQuestionReducer";

it("login test", () => {
  const props = {
    LoginResults: {
      item: { access_token: "cfcvfcccrtcc" }
    },
    match: {params: {id: 1}},
    ViewSingleQuestion: {item: { answers: [{answer: "it is that boy",
    answer_date: "Tue, 02 Oct 2018 00:00:00 GMT",
    answer_id: 1,
    question_id: 1,
    user_id: 1},{answer: "it is that boy",
    answer_date: "Tue, 02 Oct 2018 00:00:00 GMT",
    answer_id: 2,
    question_id: 1,
    user_id: 1}],
    question_date: "Tue, 02 Oct 2018 00:00:00 GMT",
    question_id: 1,
    questions: "what is andela",
    user_id: 1,
    user_name: "kegz"}},
    SingleQuestionAction: () => jest.fn()
  };

  const component = shallow(<SingleQuestion {...props} SingleQuestionAction={jest.fn()} />);
  expect(component).toMatchSnapshot();
  
});

it("Test SingleQuestion reducder", () => {
  const initialState = {
    item: {}
  };
  const action = {
    type: "SINGLE_QUESTION",
    payload: {}
  };
  expect(singleQuestionReducer(undefined, action)) !== initialState;
});

