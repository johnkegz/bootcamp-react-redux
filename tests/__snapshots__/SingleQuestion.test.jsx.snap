// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SingleQuestion)
    LoginResults={
      Object {
        "item": Object {
          "access_token": "cfcvfcccrtcc",
        },
      }
    }
    SingleQuestionAction={[MockFunction]}
    ViewSingleQuestion={
      Object {
        "item": Object {
          "answers": Array [
            Object {
              "answer": "it is that boy",
              "answer_date": "Tue, 02 Oct 2018 00:00:00 GMT",
              "answer_id": 1,
              "question_id": 1,
              "user_id": 1,
            },
            Object {
              "answer": "it is that boy",
              "answer_date": "Tue, 02 Oct 2018 00:00:00 GMT",
              "answer_id": 2,
              "question_id": 1,
              "user_id": 1,
            },
          ],
          "question_date": "Tue, 02 Oct 2018 00:00:00 GMT",
          "question_id": 1,
          "questions": "what is andela",
          "user_id": 1,
          "user_name": "kegz",
        },
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
